scripted-tests/run/backtrace/build.sbt

```diff
@@ -0,0 +1,31 @@
+import scala.sys.process.Process
+import java.util.Locale
+import scala.scalanative.build._
+
+enablePlugins(ScalaNativePlugin)
+
+scalaVersion := {
+  val scalaVersion = System.getProperty("scala.version")
+  if (scalaVersion == null)
+    throw new RuntimeException(
+      """|The system property 'scala.version' is not defined.
+         |Specify this property using the scriptedLaunchOpts -D.""".stripMargin
+    )
+  else scalaVersion
+}
+
+nativeConfig ~= { c =>
+  c.withDebugMetadata(true)
+    .withMode(Mode.debug) // otherwise, clang O2 inlines the call stack in Linux
+}
+
+lazy val debugBuild = taskKey[Unit]("Compile and run dsymutil if exists")
+debugBuild := {
+  val path = (Compile / nativeLink).value
+  if (System
+        .getProperty("os.name")
+        .toLowerCase(Locale.ROOT)
+        .startsWith("mac")) {
+    Process(s"dsymutil ${path.getAbsolutePath()}") !
```

Comment:
This step is necessary to make the positions available. Can we try to integrate it with the main workflow? We can try to run it, and then if it's not present we should log a warning for the users that dsumutil and missing and stacktraces would not contain positions, and potentially if executiono fit fails we should give another warning. 
Streaminline this into the main workflow would actually make this feature the most usefull in tests where frequent exceptions would guide us in resolving bugs and slower startup is not an issue.  

`dsymutil` is part of LLVM toolchain so we can try to discover it similary to `llvm-ar` here: https://github.com/WojciechMazur/scala-native/blob/1cdb0c71d9dcd1b527d70d0d36a0ce81855977af/tools/src/main/scala/scala/scalanative/build/LLVM.scala#L232

Reply:
> This step is necessary to make the positions available. Can we try to integrate it with the main workflow?  ...
> dsymutil is part of LLVM toolchain so we can try to discover it similary to llvm-ar here:

Ah, that sounds better! :+1: Actually `clang` also integrate `dsymutil` into its workflow.

Reply:
Just as a note, Apple toolchain does supply `dsymutil` but not `llvm-ar` but I understand `ar` will work. I am thinking the archive won't work on Apple toolchain with our current setup.